engine/runtime-compiler/src/main/scala/org/enso/compiler/pass/analyse/alias/graph/Graph.scala

```diff
@@ -93,8 +80,11 @@ sealed class Graph(
     *
     * @return a unique identifier for this graph
     */
-  def nextId(): Graph.Id = {
+  final def nextId(): Graph.Id = {
     val nextId = _nextIdCounter
+    if (nextId < 0) {
+      throw new IllegalStateException("Cannot emit new IDs. Frozen!")
```

Comment:
IMHO it is better to have the explicit `boolean frozen` field, than encoding the *frozen* state as `nextId == -1`.

Reply:
I guess, my original idea was to not serialize `nextIdCounter` in serde, but initialize it to `-1` after deserialization - e.g. freeze it. But (as freezing doesn't work - #11022) I had to give up on that idea. Now the code is ready for freezing (by assigning the counter to -1), but there is no freezing happening. Hopefully it will, one day.